src/rws_subscription.cpp

```diff
@@ -70,10 +81,12 @@ namespace abb :: rws
       }
       catch (Poco::TimeoutException const&)
       {
-        BOOST_THROW_EXCEPTION(
-          TimeoutError {"WebSocket frame receive timeout"}
-            << boost::errinfo_nested_exception(boost::current_exception())
-        );
+        if (now >= deadline)
```

Comment:
This is sloppy. We decide whether we expect the ping or the data frame to arrive first it the line with `std::min(...)`. Here we should use the same condition. Maybe set a variable that tells us whether we wait for a ping of for a data frame.

Reply:
I agree :+1: